### ISSUE TITLE
Update pipeline to specify `--packages` option for previously source built scenarios

### ISSUE THREAD

**Issue by mthalman** [dotnet/source-build#4427]
The VMR build script currently allows for previously source built packages to be provided implicitly, by storing them in the `prereqs/packages/archive` directory, or via an explicit path set with the `--packages` option. The latter is not exercised by any automation which led to a [regression](https://github.com/dotnet/source-build/issues/4423) that went unknown for a while.

The validation pipeline does exercise the scenario of using previously source built packages, but this is done by explicitly copying the packages to the well-known `prereqs/packages/archive` location. This avoids the need to specify the `--packages` option to the build script. The problem with that is there is no test coverage of the `--packages` option.

The proposal here is to update the pipeline so that the previously source built scenarios will explicitly set the `--packages` option to a location other than `prereqs/packages/archive`.

**Comment by ViktorHofer** [comment]
Agreed. Does anyone (devs) use the well-known location today? If not, I would remove that and always require bits to be passed-in explicitly via the `--with-packages` option. Related: https://github.com/dotnet/source-build/issues/4422
